Desctiption: use packaged version of mmg3d instead of shipped with tarball,
 when it appears in Debian. See #651438
Author: Christophe Trophime <christophe.trophime@lncmi.cnrs.fr>
Reviewed-by: Anton Gladky <gladky.anton@gmail.com>
Last-Update: 2012-01-16

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -548,8 +548,18 @@
   endif(ENABLE_BAMG)
 
   if(ENABLE_MMG3D)
-    add_subdirectory(contrib/mmg3d)
-    include_directories(contrib/mmg3d/build/sources)
+    find_path(MMG3D_INC "libmmg3d.h" PATH_SUFFIXES src include)
+    if(MMG3D_INC)
+       list(APPEND EXTERNAL_INCLUDES ${MMG3D_INC})
+       find_library(MMG3D_LIB mmg3dlib4.0 PATH_SUFFIXES lib)
+       if(MMG3D_LIB)
+           list(APPEND EXTERNAL_LIBRARIES ${MMG3D_LIB})
+       endif(MMG3D_LIB)
+    else(MMG3D_INC)
+       message(STATUS "MMG3D " ${MMG3D_INC} " not Found - using contrib/mmg3d instead")
+       add_subdirectory(contrib/mmg3d)
+       include_directories(contrib/mmg3d/build/sources)
+    endif(NOT MMG3D_INC)
     set_config_option(HAVE_MMG3D "Mmg3d")
   endif(ENABLE_MMG3D)
 
